### ISSUE TITLE
[Laravel 5] Bad query when loading jobs from queue

### ISSUE THREAD

**Issue by JavoByte** [yajra/laravel-oci8#76]
I developed an application fully tested in MySQL and now I'm trying to migrate to Oracle using this package. However, I'm using a queue and when I run `php artisan queue:listen` it fails with this error:

```
  [yajra\Pdo\Oci8\Exceptions\Oci8Exception]                                                                                                                                                                                                       
  Error Code    : 907                                                                                                                                                                                                                             
  Error Message : ORA-00907: missing right parenthesis                                                                                                                                                                                            
  Position      : 173                                                                                                                                                                                                                             
  Statement     : select t2.* from ( select rownum AS "rn", t1.* from (select * from jobs where queue = :autoparam0 and reserved = :autoparam1 and available_at <= :autoparam2 order by id asc for update) t1 ) t2 where t2."rn" between 1 and 1  
  Bindings      : [default,0,1437108182] 
```

Is there anyway to fix this?

**Comment by yajra** [comment]
It seems like the issue is on `order by id asc for update`. Can you please provide some info how to reproduce this SQL? Haven't used **for update** on any of my queries yet.

**Comment by yajra** [comment]
I think this is connected on db locking. But I haven't used it IRL so any information to reproduce the error would help. Thanks!

**Comment by JavoByte** [comment]
This is the code in the Laravel framework that fails:

```
$this->database->table($this->table)
	->lockForUpdate()
	->where('queue', $this->getQueue($queue))
	->where('reserved', 0)
	->where('available_at', '<=', $this->getTime())
	->orderBy('id', 'asc')
	->first();
```

Hope it helps.

**Comment by yajra** [comment]
@JavoByte, it seems like `->lockForUpdate()` is buggy atm. The fastest solution is temporarily remove the locking. Will have to study the locking specs to be able to fix this. Sorry for the inconvenience.

**Comment by yajra** [comment]
@JavoByte, can you please verify if issue was fixed on v2.4.0. FYI: `sharedLock` is not yet supported atm in case you are using it too. Thanks!

**Comment by JavoByte** [comment]
Yep issue solved on v2.4.0. Thanks!
